### PR TITLE
LPS-116934 [A/B Testing] Use flush instead the callback of the transactional commit.

### PR DESCRIPTION
**Pull request by cgoncas**
**Description**

Not all variants are deleted from the database after removing AB Test

**Steps to reproduce it**

1. Create segment seg1
2. Create page cp1
3. Create exp1 with seg1
4. Create A/B Test test1 with exp1
5. Create variants v1 and v2
6. Delete test
7. There are still some experiences in the DB (select * from lportal.SegmentsExperience;) and also exp1 can't be moved up the default experience.

**Proposed solution**

A transactional exception is being thrown because we are updating the experiences (changing the priority of the ones left) at the same time we are deleting them.

So we need to flush the deleted experience, and run all the changes in only one transaction.

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
Hi @cgoncas, Is this commit https://github.com/liferay-tango/liferay-portal/pull/820/commits/1b414f94a93e108292da3f560be0a7ad841c2ff7 must be included in the PR?

**Comment by cgoncas** [comment]
Yeahh!!, all the commits should be included in this issue.

**Comment by cgoncas** [comment]
Yeahh!!, the right behaviour for the code is described in the test message. This code is broken right now.

**Comment by darquesdev** [comment]
It's a bit confusing for, to have check the previous test commit msg to understand what the current is doing, don't you think?

**Comment by cgoncas** [comment]
I think there is not eadier way of understand what the expected behaviour is, that show an example (automated) using a test.

**Comment by darquesdev** [comment]
The best 2 commits are self-explanatory: [1st the behaviour](url) and then the solution. But with the other (https://github.com/liferay-tango/liferay-portal/pull/820/commits/323f3fd431c69527231d7359ad896a4ebbcc4536 & https://github.com/liferay-tango/liferay-portal/pull/820/commits/1b414f94a93e108292da3f560be0a7ad841c2ff7)two it's not easy to understand

**Review by darquesdev — APPROVED** [review]
LGTM! I just left a comment with the commits msg, but after reviewing the code It's clear for. Up to you

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java [hunk 1]
```diff
@@ -203,23 +201,42 @@ public SegmentsExperience deleteSegmentsExperience(
...
+				segmentsExperiences, lowestPriority - 1, -1);
 
-		for (SegmentsExperience curSegmentsExperience : segmentsExperiences) {
-				curSegmentsExperience.setPriority(
-					curSegmentsExperience.getPriority() - 1);
+			segmentsExperiencePersistence.flush();
 
-				segmentsExperienceLocalService.updateSegmentsExperience(
-					curSegmentsExperience);
+			_updateSegmentExperiencesPriority(
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java:227]: Nice! All the priority updating logic using the same method ;)

**Comment by cgoncas** [comment]
ci.forward

**Comment by cgoncas** [comment]
> LGTM! I just left a comment with the commits msg, but after reviewing the code It's clear for. Up to you

Thanks a lot for the review, I have changed the message, let me know if this helps you to understand the fix.

**Comment by darquesdev** [comment]
Hi @cgoncas, should this PR get forwarded?

**Comment by cgoncas** [comment]
> Hi @cgoncas, should this PR get forwarded?

I am waiting your response :)

**Comment by darquesdev** [comment]
Oh, that's right! Yes, the commit messages are so good explained, thanks! please, feel free to forward it ;)

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**